homeassistant/components/rituals_perfume_genie/number.py

```diff
@@ -0,0 +1,118 @@
+"""Support for Rituals Perfume Genie numbers."""
+import logging
+from typing import Callable
+
+from pyrituals import Diffuser
+
+from homeassistant.components.number import NumberEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import ATTRIBUTES, COORDINATORS, DEVICES, DOMAIN, HUB, ROOM, SPEED
+from .entity import DiffuserEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_PERFUME_AMOUNT = 1
+MAX_PERFUME_AMOUNT = 3
+MIN_ROOM_SIZE = 1
+MAX_ROOM_SIZE = 4
+
+PERFUME_AMOUNT_SUFFIX = " Perfume Amount"
+ROOM_SIZE_SUFFIX = " Room Size"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, config_entry: ConfigEntry, async_add_entities: Callable
+) -> None:
+    """Set up the diffuser numbers."""
+    diffusers = hass.data[DOMAIN][config_entry.entry_id][DEVICES]
+    coordinators = hass.data[DOMAIN][config_entry.entry_id][COORDINATORS]
+    entities = []
+    for hublot, diffuser in diffusers.items():
+        coordinator = coordinators[hublot]
+        entities.append(DiffuserPerfumeAmount(diffuser, coordinator))
+        entities.append(DiffuserRoomSize(diffuser, coordinator))
```

Comment:
Perfume amount I can understand. At times one probably wants a different "intensity".

However, I assume, in most cases, the devices will mostly fixed in their rooms. Shouldn't room size be an integration entry option instead?

Reply:
I don't know what the effect of the room size is on the operation of the device. Technically it's just a jar of perfume blown over by a fan. Adjusting the strength changes the speed of the fan. I have no idea what room size can change, probably it's only used to calculate how much perfume is used.

> the devices will mostly fixed in their rooms

The first version has a battery that makes it easy to move the device to another room. In this case, it may be useful to detect the location of the device and then automatically change the room size.

I think not many users will use this entity but here is a feature request by @Pirol62 in https://github.com/fred-oranje/rituals-genie/issues/16:
> In the app, there are options to set the roomsize in m2 and the intensity (low, medium and high)
> Then it would be great to have them in HA as well.

Reply:
Room size must be set to 15, 30, 60, or 100 in the app. In the API this is 1, 2, 3 and 4 respectively and this is how it is now in this PR. I think it would be better to show the values in square meters. Should this be done with the number or input_select entity?

Reply:
`input_select` entities are not usable for integrations.

So I guess that makes the currently solution in this PR the most logical one than 👍